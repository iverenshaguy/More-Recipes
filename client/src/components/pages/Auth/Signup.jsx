import React from 'react';
import PropTypes from 'prop-types';
import Form from '../../shared/Forms';

/**
 * @exports
 * @function Signup
 * @param {object} props
 * @returns {JSX} Signup
 */
const Signup = (props) => {
  const signupMeta = {
    title: 'Register for a New Account',
    btnText: 'SIGN UP',
    extra: <p className="text-center">Already have an account, <a href="/login" onClick={e => props.changeForm(e, 'login')}>Log in here.</a></p>
  };

  return <Form {...props} type="signup" meta={signupMeta} changeForm={props.changeForm} />;
};

Signup.propTypes = {
  changeForm: PropTypes.func.isRequired
};

export default Signup;

