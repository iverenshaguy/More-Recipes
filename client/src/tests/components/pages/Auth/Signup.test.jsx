import React from 'react';
import Signup from '../../../../components/pages/Auth/Signup';

const changeFormMock = jest.fn();


describe('Signup', () => {
  it('renders correctly', () => {
    const shallowWrapper = shallow(<Signup changeForm={changeFormMock} />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
