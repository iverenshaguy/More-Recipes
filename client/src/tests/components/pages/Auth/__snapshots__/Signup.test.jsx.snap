// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup renders correctly 1`] = `
<Connect(Form)
  changeForm={[Function]}
  meta={
    Object {
      "btnText": "SIGN UP",
      "extra": <p
        className="text-center"
      >
        Already have an account, 
        <a
          href="/login"
          onClick={[Function]}
        >
          Log in here.
        </a>
      </p>,
      "title": "Register for a New Account",
    }
  }
  type="signup"
/>
`;
